File: doc/tutoNLP/loops/l3_all_same_name.ml
open Optitrust
open Prelude

(* Difficulty: Level 3
   Request: target every loop named x in blur
   Expected target: [nbMulti; cFunDef "blur"; cFor "x"]
   Accepted variant: [nbMulti; cFunBody "blur"; cFor "x"] *)

let _ = Run.script_cpp (fun () ->
  !! Show.target [nbMulti; cFunDef "blur"; cFor "x"];
)

File: doc/tutoNLP/loops/l3_all_same_name.cpp
void blur(float* out) {
  for (int x = 0; x < 4; x++) {
    out[x] = 0.0f;
  }
  for (int y = 0; y < 4; y++) {
    for (int x = 0; x < 4; x++) {
      out[y * 4 + x] += 1.0f;
    }
  }
}
